test/e2e_node/oomkiller_linux_test.go

```diff
@@ -59,13 +82,31 @@ var _ = SIGDescribe("OOMKiller [LinuxOnly] [NodeConformance]", func() {
 		})
 	}
 	for _, tc := range testCases {
-		runOomKillerTest(f, tc)
+		runOomKillerTest(f, tc, 0)
 	}
 })
 
-func runOomKillerTest(f *framework.Framework, testCase testCase) {
+func runOomKillerTest(f *framework.Framework, testCase testCase, kubeReservedMemory float64) {
 	ginkgo.Context(testCase.name, func() {
+		var oldCfg *kubeletconfig.KubeletConfiguration
+		var err error
 		ginkgo.BeforeEach(func() {
+			// Update KubeReservedMemory in KubeletConfig.
+			if kubeReservedMemory > 0 {
+				oldCfg, err = getCurrentKubeletConfig(context.TODO())
+				framework.ExpectNoError(err)
+				newCfg := oldCfg.DeepCopy()
+				if newCfg.KubeReserved == nil {
+					newCfg.KubeReserved = map[string]string{}
+				}
+				// There's a race condition observed between system OOM and cgroup OOM if node alocatable
+				// memory is equal to node capacity. Hence, reserving a fraction of node's memory capacity for
+				// K8s components such that node allocatable memory is less than node capacity to
+				// observe OOM kills at cgroup level instead of system OOM kills.
+				newCfg.KubeReserved["memory"] = fmt.Sprintf("%d", int(kubeReservedMemory*getLocalNode(context.TODO(), f).Status.Capacity.Memory().AsApproximateFloat64()))
+				updateKubeletConfig(context.TODO(), f, newCfg, true)
```

Comment:
small nit: I would recommend to use the `tempSetCurrentKubeletConfig` which will automatically revert the kubelet config in the after each (that way you don't need to store the old config or worry about reverting to it).

e.g. see https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/podresources_test.go#L809-L819